File: 09-Maximize Happiness of Selected Children/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end(), greater<int>());
        
        long long res = 0;
        int counter = 0;

        for (int i = 0; i < min(k, static_cast<int>(happiness.size())); ++i) {
            res += max(0, happiness[i] - counter);
            ++counter;
        }
        return res;
    }
};

File: 09-Maximize Happiness of Selected Children/solution.py
class Solution:
    def maximumHappinessSum(self, happiness, k):
        happiness.sort(reverse=True)

        res = 0
        counter = 0
        for i in range(min(k, len(happiness))):
            res += max(0, happiness[i] - counter)
            counter += 1
        
        return res
